components/eventing-controller/controllers/backend/reconciler.go

```diff
@@ -1000,6 +1000,10 @@ func (r *Reconciler) getMutatingAndValidatingWebHookConfig(ctx context.Context)
 	return &mutatingWH, &validatingWH, nil
 }
 
+func (r *Reconciler) isOauth2CredentialsInitialized() bool {
+	return len(r.credentials.clientID) > 0 && len(r.credentials.clientSecret) > 0
```

Comment:
Does it make sense to also check the tokenURL and certsURL?

Reply:
Also please write a unit test for this function.

Reply:
Not sure atm, that `certsURL` would be non-empty when oauth2 feature flag is disabled. Thats why I only checked the fields which are used in both cases (i.e. oauth2 feature flag enabled/disabled)

Reply:
I will add unit test in the next PR.

Reply:
Please also check the tokenURL and certsURL (if possible) in the followup PR.